// @deeplint/deepscanner-aws/lib/aws/helper/aws-region-helper.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.awsRegions = exports.AWSRegionsHelper = void 0;
const lodash_1 = require("lodash");
class AWSRegionsHelper {
    static getServiceRegions(serviceName) {
        if (lodash_1.has(exports.awsRegions, serviceName)) {
            return lodash_1.get(exports.awsRegions, serviceName);
        }
        return ['us-east-1'];
    }
}
exports.AWSRegionsHelper = AWSRegionsHelper;
exports.awsRegions = {
    ACM: [
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-east-1',
        'us-east-2',
        'us-gov-west-1',
        'us-west-1',
        'us-west-2',
    ],
    ACMPCA: [
        'ap-northeast-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'us-east-1',
        'us-east-2',
        'us-west-2',
    ],
    APIGateway: [
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-east-1',
        'us-east-2',
        'us-west-1',
        'us-west-2',
    ],
    AppStream: [
        'us-east-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'eu-central-1',
        'eu-west-1',
    ],
    AppSync: [
        'us-east-2',
        'us-east-1',
        'us-west-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ap-northeast-1',
        'eu-central-1',
        'eu-west-1',
        'us-east-2',
        'us-east-1',
        'us-west-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ap-northeast-1',
        'eu-central-1',
        'eu-west-1',
    ],
    ApplicationAutoScaling: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
    ],
    AutoScaling: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    AutoScalingPlans: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    Batch: [
        'us-west-2',
        'us-west-1',
        'us-east-2',
        'us-east-1',
        'ca-central-1',
        'eu-west-2',
        'eu-west-1',
        'eu-central-1',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-south-1',
        'ap-southeast-2',
        'ap-southeast-1',
        'sa-east-1',
    ],
    Cloud9: ['us-east-2', 'us-east-1', 'us-west-2', 'ap-southeast-1', 'eu-west-1'],
    CloudDirectory: [
        'us-east-2',
        'us-east-1',
        'us-west-2',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
    ],
    CloudFormation: [
        'us-east-1',
        'us-east-2',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'eu-central-1',
        'sa-east-1',
        'us-gov-west-1',
    ],
    CloudHSM: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'eu-central-1',
        'eu-west-1',
        'us-gov-west-1',
    ],
    CloudHSMV2: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'eu-central-1',
        'eu-west-1',
        'us-gov-west-1',
    ],
    CloudSearch: [
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-southeast-1',
        'ap-southeast-2',
        'eu-central-1',
        'eu-west-1',
        'sa-east-1',
    ],
    CloudSearchDomain: [
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-southeast-1',
        'ap-southeast-2',
        'eu-central-1',
        'eu-west-1',
        'sa-east-1',
    ],
    CloudTrail: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    CloudWatch: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    CloudWatchEvents: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    CloudWatchLogs: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    CodeBuild: [
        'us-east-1',
        'us-east-2',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'eu-central-1',
        'sa-east-1',
    ],
    CodeCommit: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
    ],
    CodeDeploy: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    CodePipeline: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
    ],
    CodeStar: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
    ],
    CognitoSync: [
        'us-east-2',
        'us-east-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
    ],
    Comprehend: ['us-east-2', 'us-east-1', 'us-west-2', 'ap-southeast-2', 'eu-west-1'],
    Connect: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    DMS: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    DataPipeline: ['us-east-1', 'us-west-2', 'ap-northeast-1', 'ap-southeast-2', 'eu-west-1'],
    DeviceFarm: ['us-west-2'],
    DirectConnect: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    DynamoDB: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    DynamoDBStreams: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    EC2: [
        'eu-north-1',
        'ap-south-1',
        'eu-west-3',
        'eu-west-2',
        'eu-west-1',
        'ap-northeast-2',
        'ap-northeast-1',
        'sa-east-1',
        'ca-central-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'eu-central-1',
        'us-east-1',
        'us-east-2',
        'us-west-1',
        'us-west-2',
    ],
    ECR: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    ECS: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    EFS: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-2',
        'ap-southeast-1',
        'ap-northeast-1',
        'eu-central-1',
        'eu-west-1',
        'ap-southeast-2',
    ],
    ELB: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ca-central-1',
        'ap-south-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-southeast-1',
        'ap-southeast-2',
        'ap-northeast-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
    ],
    ELBv2: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ca-central-1',
        'ap-south-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-southeast-1',
        'ap-southeast-2',
        'ap-northeast-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
    ],
    EMR: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    ES: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    ElastiCache: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    ElasticBeanstalk: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ca-central-1',
        'ap-south-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-southeast-1',
        'ap-southeast-2',
        'ap-northeast-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
    ],
    ElasticTranscoder: [
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ap-northeast-1',
        'eu-west-1',
    ],
    Firehose: [
        'us-west-2',
        'eu-west-3',
        'us-east-1',
        'ap-south-1',
        'us-east-2',
        'ca-central-1',
        'eu-west-1',
        'eu-central-1',
        'sa-east-1',
        'ap-northeast-2',
        'eu-west-2',
        'ap-northeast-1',
        'us-west-1',
        'ap-southeast-1',
        'ap-southeast-2',
    ],
    GameLift: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-south-1',
        'ap-northeast-2',
        'ap-southeast-1',
        'ap-southeast-2',
        'ap-northeast-1',
        'ca-central-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'sa-east-1',
    ],
    Glacier: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'us-gov-west-1',
    ],
    Glue: [
        'us-east-2',
        'us-east-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
    ],
    Greengrass: ['us-east-1', 'us-west-2', 'ap-southeast-2', 'ap-northeast-1', 'eu-central-1', 'eu-west-1'],
    GuardDuty: [
        'ap-south-1',
        'ap-northeast-2',
        'ap-southeast-1',
        'ap-southeast-2',
        'ap-northeast-1',
        'ca-central-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'us-east-1',
        'us-east-2',
        'us-west-1',
        'us-west-2',
        'sa-east-1',
    ],
    Inspector: [
        'us-east-1',
        'us-east-2',
        'us-west-1',
        'us-west-2',
        'ap-south-1',
        'ap-northeast-2',
        'ap-southeast-2',
        'ap-northeast-1',
        'eu-west-1',
        'eu-central-1',
        'us-gov-west-1',
    ],
    IoTAnalytics: ['us-east-1', 'us-east-2', 'us-west-2', 'eu-west-1'],
    KMS: [
        'us-east-1',
        'us-east-2',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    Kinesis: ['us-east-1', 'us-west-2', 'ap-northeast-1', 'eu-central-1', 'eu-west-1'],
    KinesisAnalytics: ['us-east-1', 'us-west-2', 'ap-northeast-1', 'eu-central-1', 'eu-west-1'],
    KinesisVideo: ['us-east-1', 'us-west-2', 'ap-northeast-1', 'eu-central-1', 'eu-west-1'],
    KinesisVideoArchivedMedia: ['us-east-1', 'us-west-2', 'ap-northeast-1', 'eu-central-1', 'eu-west-1'],
    KinesisVideoMedia: ['us-east-1', 'us-west-2', 'ap-northeast-1', 'eu-central-1', 'eu-west-1'],
    Lambda: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ap-northeast-1',
        'ca-central-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'cn-north-1',
        'cn-northwest-1',
    ],
    Lightsail: [
        'us-east-2',
        'us-east-1',
        'us-west-2',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'ca-central-1',
        'ap-south-1',
        'ap-northeast-2',
        'ap-southeast-1',
        'ap-southeast-2',
        'ap-northeast-1',
    ],
    MQ: ['us-east-2', 'us-east-1', 'us-west-2', 'ap-southeast-2', 'eu-central-1', 'eu-west-1'],
    MachineLearning: ['us-east-1', 'eu-west-1'],
    Macie: ['us-east-1', 'us-west-2'],
    MediaConvert: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'sa-east-1',
    ],
    MediaLive: [
        'us-east-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'eu-central-1',
        'eu-west-1',
        'sa-east-1',
    ],
    MediaPackage: [
        'us-east-1',
        'us-west-2',
        'ap-northeast-2',
        'ap-southeast-1',
        'ap-southeast-2',
        'ap-northeast-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-3',
        'sa-east-1',
    ],
    MediaStore: [
        'us-east-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-southeast-2',
        'eu-central-1',
        'eu-west-1',
    ],
    MediaStoreData: [
        'us-east-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-southeast-2',
        'eu-central-1',
        'eu-west-1',
    ],
    MediaTailor: ['us-east-1', 'us-west-2', 'ap-southeast-1', 'ap-southeast-2', 'ap-northeast-1', 'eu-west-1'],
    MigrationHub: ['us-west-2'],
    Mobile: ['us-east-1'],
    MobileAnalytics: ['us-east-1'],
    Organizations: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'sa-east-1',
    ],
    PI: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-south-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-southeast-1',
        'ap-southeast-2',
        'ap-northeast-1',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
    ],
    Pinpoint: ['us-east-1'],
    Polly: [
        'us-east-1',
        'us-east-2',
        'us-west-1',
        'us-west-2',
        'ca-central-1',
        'eu-west-1',
        'eu-central-1',
        'eu-west-2',
        'eu-west-3',
        'ap-southeast-1',
        'ap-southeast-2',
        'ap-northeast-2',
        'ap-northeast-1',
        'ap-south-1',
        'sa-east-1',
    ],
    RDS: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    Redshift: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    Rekognition: [
        'us-east-2',
        'us-east-1',
        'us-west-2',
        'ap-southeast-2',
        'ap-northeast-1',
        'eu-west-1',
        'ap-northeast-2',
        'ap-south-1',
    ],
    ResourceGroups: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-south-1',
        'ap-northeast-2',
        'ap-southeast-1',
        'ap-southeast-2',
        'ap-northeast-1',
        'ca-central-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
    ],
    ResourceGroupsTaggingAPI: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-south-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-southeast-1',
        'ap-southeast-2',
        'ap-northeast-1',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
    ],
    S3: ['us-east-1', 'us-east-2', 'us-west-1', 'us-west-2'],
    SES: ['us-east-1', 'us-west-2', 'eu-west-1', 'us-east-1', 'us-west-2', 'eu-west-1'],
    SMS: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    SNS: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    SQS: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-northeast-3',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'cn-northwest-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    SageMaker: [
        'us-east-1',
        'us-east-2',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-southeast-2',
        'eu-central-1',
        'eu-west-1',
    ],
    SageMakerRuntime: [
        'us-east-1',
        'us-east-2',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-southeast-2',
        'eu-central-1',
        'eu-west-1',
    ],
    SecretsManager: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
    ],
    ServerlessApplicationRepository: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'sa-east-1',
    ],
    ServiceCatalog: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-south-1',
        'ap-northeast-2',
        'ap-southeast-1',
        'ap-southeast-2',
        'ap-northeast-1',
        'ca-central-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
    ],
    Shield: ['us-east-1'],
    SimpleDB: [
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-southeast-1',
        'ap-southeast-2',
        'ap-northeast-1',
        'eu-west-1',
        'sa-east-1',
    ],
    Snowball: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ca-central-1',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ap-northeast-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
    ],
    StepFunctions: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'us-gov-west-1',
    ],
    StorageGateway: [
        'us-east-2',
        'us-east-1',
        'us-west-1',
        'us-west-2',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-south-1',
        'ap-southeast-1',
        'ap-southeast-2',
        'ca-central-1',
        'cn-north-1',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'eu-west-3',
        'sa-east-1',
        'us-gov-west-1',
    ],
    Translate: ['us-east-2', 'us-east-1', 'us-west-2', 'eu-west-1', 'us-gov-west-1'],
    WorkDocs: ['us-east-1', 'us-west-2', 'ap-southeast-1', 'ap-southeast-2', 'ap-northeast-1', 'eu-west-1'],
    WorkMail: [
        'us-east-1',
        'us-east-1',
        'us-east-1',
        'us-east-1',
        'us-east-1',
        'us-east-1',
        'us-east-1',
        'us-west-2',
        'us-west-2',
        'us-west-2',
        'us-west-2',
        'us-west-2',
        'us-west-2',
        'us-west-2',
        'eu-west-1',
        'eu-west-1',
        'eu-west-1',
        'eu-west-1',
        'eu-west-1',
        'eu-west-1',
        'eu-west-1',
    ],
    WorkSpaces: [
        'us-east-1',
        'us-west-2',
        'ca-central-1',
        'ap-northeast-1',
        'ap-northeast-2',
        'ap-southeast-1',
        'ap-southeast-2',
        'eu-central-1',
        'eu-west-1',
        'eu-west-2',
        'sa-east-1',
    ],
};
