// @deeplint/deepscanner-aws/lib/aws/resources/ec2-eip.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EC2EIPProvider = void 0;
const aws_error_handler_1 = require("../helper/aws-error-handler");
const AWS = require("aws-sdk");
const aws_provider_1 = require("../helper/aws-provider");
const _ = require("lodash");
class EC2EIPProvider extends aws_provider_1.AwsProvider {
    collect(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.listAllAddresses(context);
        });
    }
    listAllAddresses(context) {
        return __awaiter(this, void 0, void 0, function* () {
            const result = [];
            const serviceName = 'EC2';
            const regions = _.has(context, 'inputs.regions') ? context.inputs.regions : this.getRegions(serviceName);
            try {
                for (const region of regions) {
                    AWS.config.update({ region: region });
                    const ec2 = this.getClient(serviceName, region);
                    const eipAddressData = yield ec2.describeAddresses().promise();
                    if (eipAddressData && eipAddressData.Addresses) {
                        for (const address of eipAddressData.Addresses) {
                            if (address.PublicIp) {
                                result.push({
                                    name: address.PublicIp,
                                    type: EC2EIPProvider.RESOURCE_TYPE,
                                    meta: {
                                        region: region,
                                    },
                                    properties: Object.assign({}, address),
                                });
                            }
                        }
                    }
                }
            }
            catch (error) {
                aws_error_handler_1.handle(error);
            }
            return result;
        });
    }
}
exports.EC2EIPProvider = EC2EIPProvider;
EC2EIPProvider.RESOURCE_TYPE = 'aws::ec2::eip';
