// @deeplint/deepscanner-aws/lib/aws/resources/s3-bucket.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.S3BucketProvider = void 0;
const aws_error_handler_1 = require("../helper/aws-error-handler");
const aws_provider_1 = require("../helper/aws-provider");
const _ = require("lodash");
class S3BucketProvider extends aws_provider_1.AwsProvider {
    collect(context) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.listAllBuckets(context);
        });
    }
    listAllBuckets(context) {
        return __awaiter(this, void 0, void 0, function* () {
            const result = [];
            const serviceName = 'S3';
            const regions = _.has(context, 'inputs.regions') ? context.inputs.regions : this.getRegions(serviceName);
            try {
                for (const region of regions) {
                    const s3 = this.getClient(serviceName, region);
                    const s3BucketsData = yield s3.listBuckets().promise();
                    if (s3BucketsData && s3BucketsData.Buckets) {
                        for (const bucket of s3BucketsData.Buckets) {
                            if (bucket.Name) {
                                const versioning = yield s3
                                    .getBucketVersioning({ Bucket: bucket.Name })
                                    .promise();
                                const encryption = yield s3
                                    .getBucketEncryption({ Bucket: bucket.Name })
                                    .promise();
                                const policyStatus = yield s3
                                    .getBucketPolicyStatus({ Bucket: bucket.Name })
                                    .promise();
                                result.push({
                                    name: bucket.Name,
                                    type: S3BucketProvider.RESOURCE_TYPE,
                                    meta: {
                                        region: region,
                                    },
                                    properties: Object.assign(Object.assign(Object.assign({}, versioning), encryption), policyStatus),
                                });
                            }
                        }
                    }
                }
            }
            catch (error) {
                aws_error_handler_1.handle(error);
            }
            return result;
        });
    }
}
exports.S3BucketProvider = S3BucketProvider;
S3BucketProvider.RESOURCE_TYPE = 'aws::s3::bucket';
